File: test/services/markdown/MarkdownConverterTest.java
package services.markdown;

import com.petebevin.markdown.MarkdownProcessor;
import org.junit.Test;

import static org.mockito.Mockito.*;

/**
 * Date: 01.02.2012 at 21:43
 *
 * @author Marcin Swierczynski
 */
public class MarkdownConverterTest {

	@Test
	public void contentProcessed() throws Exception {
		MarkdownConverter markdownConverterSpy = spy(new MarkdownConverter());

		MarkdownProcessor markdownProcessorMock = mock(MarkdownProcessor.class);
		when(markdownConverterSpy.createMarkdownProcessor()).thenReturn(markdownProcessorMock);

		String content = "content to convert";
		markdownConverterSpy.convert(content);
		
		verify(markdownProcessorMock).markdown(content);
	}

}

File: test/services/markdown/MarkdownConverterTest.scala
package services.markdown

import org.specs2.mutable.Specification
import org.specs2.mock._
import com.petebevin.markdown.MarkdownProcessor

/**
 * Date: 24.03.2012 at 14:11
 *
 * @author Marcin Swierczynski
 */

class MarkdownConverterTest extends Specification with Mockito {

	"calls markdown processor" in {
		val markdownConverterComponent = new MarkdownConverterComponent {
			val markdownConverter: MarkdownConverter = spy(new MarkdownConverter)
		}

		val markdownProcessorMock: MarkdownProcessor = mock[MarkdownProcessor]
		markdownConverterComponent.markdownConverter.createMarkdownProcessor() returns markdownProcessorMock

		val content = "Content to convert"
		markdownConverterComponent.markdownConverter.convert(content)

		there was one(markdownProcessorMock).markdown(content)
	}

}
